tools/Repo-Tasks.psm1

```diff
@@ -0,0 +1,245 @@
+        [Parameter(ParameterSetName='UserIdParamSet', Mandatory=$true, HelpMessage = "SubscriptionId you would like to use")]
+        [ValidateNotNullOrEmpty()]
+        [string]$SubscriptionId,
+
+        [Parameter(ParameterSetName='SpnParamSet', Mandatory=$true, HelpMessage='AADTenant/TenantId you would like to use')]
+        [ValidateNotNullOrEmpty()]
+        [string]$TenantId,
+
+        [ValidateSet("Playback", "Record", "None")]
+        [string]$RecordMode='Playback',
+
+        [ValidateSet("Prod", "Dogfood", "Current", "Next")]
+        [string]$TargetEnvironment='Prod'
+    )
+
+    [string]$uris="https://management.azure.com/"
+
+    $formattedConnStr = [string]::Format("SubscriptionId={0};HttpRecorderMode={1};Environment={2}", $SubscriptionId, $RecordMode, $TargetEnvironment)
+
+    if([string]::IsNullOrEmpty($UserId) -eq $false)
+    {
+        $formattedConnStr = [string]::Format([string]::Concat($formattedConnStr, ";UserId={0}"), $UserId)
+    }
+
+    if([string]::IsNullOrEmpty($Password) -eq $false)
+    {
+        $formattedConnStr = [string]::Format([string]::Concat($formattedConnStr, ";Password={0}"), $Password)
+    }
+
+    if([string]::IsNullOrEmpty($TenantId) -eq $false)
+    {
+        $formattedConnStr = [string]::Format([string]::Concat($formattedConnStr, ";AADTenant={0}"), $TenantId)
+    }
+
+    if([string]::IsNullOrEmpty($ServicePrincipal) -eq $false)
+    {
+        $formattedConnStr = [string]::Format([string]::Concat($formattedConnStr, ";ServicePrincipal={0}"), $ServicePrincipal)
+    }
+
+    if([string]::IsNullOrEmpty($ServicePrincipalSecret) -eq $false)
+    {
+        $formattedConnStr = [string]::Format([string]::Concat($formattedConnStr, ";ServicePrincipalSecret={0}"), $ServicePrincipalSecret)
+    }
+    
+    $formattedConnStr = [string]::Format([string]::Concat($formattedConnStr, ";BaseUri={0}"), $uris)
+
+    Write-Host "Below connection string is ready to be set"
+    Print-ConnectionString $UserId $Password $SubscriptionId $TenantId $ServicePrincipal $ServicePrincipalSecret $RecordMode $TargetEnvironment $uris
+
+    #Set connection string to Environment variable
+    $env:TEST_CSM_ORGID_AUTHENTICATION=$formattedConnStr
+    Write-Host ""
+
+    # Retrieve the environment variable
+    Write-Host ""
+    Write-Host "Below connection string was set. Start Visual Studio by typing devenv" -ForegroundColor Green
+    [Environment]::GetEnvironmentVariable($envVariableName)
+    Write-Host ""
+    
+    Write-Host "If your needs demand you to set connection string differently, for all the supported Key/Value pairs in connection string"
+    Write-Host "Please visit https://github.com/Azure/azure-powershell/blob/dev/documentation/Using-Azure-TestFramework.md" -ForegroundColor Yellow
+}
+
+Function Print-ConnectionString([string]$uid, [string]$pwd, [string]$subId, [string]$aadTenant, [string]$spn, [string]$spnSecret, [string]$recordMode, [string]$targetEnvironment, [string]$uris)
```

Comment:
I actually liked your earlier suggestion that, this will be an helpful function for users to look at their connection string.